sentinelhub/api/batch.py

```diff
@@ -172,68 +166,68 @@ def iter_tiling_grids(self, **kwargs):
         return self.client.get_json(self._get_tiling_grids_url(grid_id), use_session=True)
 
-    def iter_requests(self, user_id=None, search=None, sort=None, **kwargs):
+    def iter_requests(
+        self, user_id: Optional[str] = None, search: Optional[str] = None, sort: Optional[str] = None, **kwargs: Any
+    ) -> Iterator["BatchRequest"]:
         """Iterate existing batch requests
 
         `Batch API reference
         <https://docs.sentinel-hub.com/api/latest/reference/#operation/getAllBatchProcessRequests>`__
 
         :param user_id: Filter requests by a user id who defined a request
-        :type user_id: str or None
         :param search: A search query to filter requests
-        :type search: str or None
         :param sort: A sort query
-        :type sort: str or None
         :param kwargs: Any additional parameters to include in a request query
         :return: An iterator over existing batch requests
-        :rtype: Iterator[BatchRequest]
         """
         params = remove_undefined({"userid": user_id, "search": search, "sort": sort, **kwargs})
         feature_iterator = SentinelHubFeatureIterator(
             client=self.client, url=self._get_process_url(), params=params, exception_message="No requests found"
         )
         for request_info in feature_iterator:
-            yield BatchRequest.from_dict(request_info)
+            yield BatchRequest.from_dict(request_info)  # type: ignore[attr-defined]
 
-    def get_latest_request(self):
+    def get_latest_request(self) -> "BatchRequest":
         """Provides a batch request that has been created the latest
 
         :return: Batch request info
-        :rtype: BatchRequest
         """
         latest_request_iter = self.iter_requests(sort="created:desc", count=1)
         try:
             return next(latest_request_iter)
         except StopIteration as exception:
             raise ValueError("No batch request is available") from exception
 
-    def get_request(self, batch_request):
+    def get_request(self, batch_request: BatchRequestType) -> "BatchRequest":
         """Collects information about a single batch request
 
         `Batch API reference
         <https://docs.sentinel-hub.com/api/latest/reference/#operation/getSingleBatchProcessRequestById>`__
 
         :return: Batch request info
-        :rtype: BatchRequest
         """
         request_id = self._parse_request_id(batch_request)
         request_info = self.client.get_json(url=self._get_process_url(request_id), use_session=True)
-        return BatchRequest.from_dict(request_info)
-
-    def update_request(self, batch_request, output=None, description=None, **kwargs):
+        return BatchRequest.from_dict(request_info)  # type: ignore[attr-defined]
+
+    def update_request(
+        self,
+        batch_request: BatchRequestType,
+        output: Optional[Dict[str, Any]] = None,
+        description: Optional[str] = None,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
```

Comment:
That's true, in the decoding process we sort-of extend JSON output type to also support an empty string. Suggestions:

- `ExtendedJsonType`
- I think we could change returning an empty string into returning `None` without breaking too much code. In that case it can become `Optional[JsonType]`. :thinking:

Reply:
I like the `Optional[JsonType]` idea

Reply:
Ok, let's then go to `Optional[JsonType]`. You'll have to make a minor change in `decode_data` function to return `None`. But as far as I can tell, this shouldn't break anything in our codebase.

Reply:
As agreed, this will be addressed in the next PRs.